### DISCUSSION TITLE
Disable default this field is required behaviour...

### DISCUSSION THREAD

**Discussion by WingsDevelopment** [react-hook-form/react-hook-form#8263] [General] [answered]
How do i disable default html message 'this field is required' and show error below the field?
Show this:
![image](https://user-images.githubusercontent.com/24592343/165740872-b32a34cd-d0dd-4e6b-aa4a-24089498c5d5.png)

Instead of this:
![image](https://user-images.githubusercontent.com/24592343/165740789-b1d76750-1f25-435b-9682-46cd6d4432b2.png)

So far I have a code for input field:

```
<Controller
            control={control}
            name="amount"
            render={({ field: { onChange, value } }) => (
              <Form.Control
                type="text"
                value={value ? value : ""}
                placeholder={t("Maximum")}
                {...(register("amount"), { required: true })}
                onChange={(e) => onChange(e.target.value)}
              />
            )}
          />

          {errors?.amount?.type === "required" && (
            <span className="validation-error">
              {t("This field is required")}
            </span>
          )}
```

**Comment by Moshyfawn** [comment]
Can you provide a simple codesandbox with the reproduction?

**Reply by WingsDevelopment** [reply]
https://codesandbox.io/s/fervent-sinoussi-mwsynb?file=/src/App.tsx

**Comment by WingsDevelopment** [comment]
Here is a codesandbox:
https://codesandbox.io/s/fervent-sinoussi-mwsynb?file=/src/App.tsx

**Comment by Moshyfawn** [comment, marked as answer]
Ah, it's [the default behavior for RB](https://react-bootstrap.github.io/forms/validation/#native-html5-form-validation), I believe.

You can disable the native HTML form validation by providing the `noValidate` attribute to your `<Form />` component

**Reply by WingsDevelopment** [reply]
But then validation is disabled? Lets forget bootstrap for a second, I didn't wanna use bootstrap validation..
I've updated codesandbox with noValidate and I have empty errors object, why?
Can you check update? 
https://codesandbox.io/s/fervent-sinoussi-mwsynb?file=/src/App.tsx:560-570

**Reply by WingsDevelopment** [reply]
I've removed bootstrap form and input field..

**Reply by Moshyfawn** [reply]
You are mixing both controlled and uncontrolled input APIs _(`Controller` & `register`)_.

To provide validation rules to a controlled component, use `rules` prop on the `<Controller />` component instead.

```jsx
<Controller
  rules={{
    required: true
  }}
/>
```

See [this code example](https://codesandbox.io/s/fast-sun-u51xgs?file=%2Fsrc%2FApp.tsx%3A997-1002)

**Reply by WingsDevelopment** [reply]
Oh I see, now I understand. Thanks, I got it working with rules={{ required: true }} on Controller.
